### PR TITLE
PA tier cost rebalance

### PR DESCRIPTION
**Pull request by Tostov**
## Changes

Increased power armor tier apparel production costs. Generally increased helmet mass a bit.
Increased the avalibility of money for elite mercs so they can still reasonably get said armor
Tweaked some of the existing patches.
Re-did a lot of the rimsenal armor specifics to more closely match some of the vanilla attributes, as well as fix some instances where the armors had practically vanilla mass/boosting stats.
Reduction in plate and simple helmet costs.

## Reasoning

Currently, power armor provides, proportionally for its raw resource cost, significantly more protection, coverage and passive benefits than any other options. While it's reasonable for higher tech items to make more efficent use of your resources and provide higher protection levels to justify the tech investment, power armor apparels have an extreme disparity in their cost vs effectiveness as armor. Not only is power armor extremely strong, but it is also easiy the most cost efficent.

## Testing

Check tests you have performed:
- [x] Game runs without errors
- [x] (For compatibility patches) ...with and without patched mod loaded
- [x] Playtested a colony (17h 20min)

### CODE REVIEW

**Comment by Tostov** [comment]
Currently, this includes human power armor apparel that I know of outside of the the vanilla armor expaneded stuff and the spartan foundry stuff since both have had changes. The intent is to also do this for alien power armor apparel as well but I'd like some feedback on how the current suggested numbers look.

**Comment by heavenisaplaceonearth1** [comment]
I like the buffs to rimsenal armor; I always thought that they needed more carry weight and a little more armor. I did like the additional resources in that it made me more proactive in investing in other combat stuff like smoke grenades or flanking, since it took time to procure the needed devilstrand or plasteel. I don't know if the cost rebalance is a bit too much or just right, but I do think that it's a step in the right direction.

Would you mind going over Jin roh armor as well, I like them a lot but I think that they could cost a bit more since they are so effective.

**Comment by Tostov** [comment]
> Would you mind going over Jin roh armor as well, I like them a lot but I think that they could cost a bit more since they are so effective.

Jin Roh patch is external. After we resolve the changes it's probably reasonable to bug mile about it, or patch the costs, but since it's external it's probably better to leave negotiating that till after we have solid numbers here.

**Comment by Breadbox2k19** [comment]
The changes to PA are good, the reduction in cost in other forms of protection is uneeded as they are not underpowered at all. They only look bad due to how absurdly OP power armours are.

**Comment by Tostov** [comment]
> The changes to PA are good, the reduction in cost in other forms of protection is uneeded as they are not underpowered at all. They only look bad due to how absurdly OP power armours are.

This is incorrect, both are also stupidly expensive compared to conventional armor vests.

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
Armor cost is a bit of a conundrum; on the one hand, it's extremely important with CE--far more so than in vanilla. However, on the other hand, without a vanilla way to repair apparel, it's semi-disposable and has to be balanced as such.

Additionally, increasing the crafting costs has diminishing returns, especially for late game items such as power armor. By the point that a colony is seriously looking at mass producing power armor, an extra 50 - 60 plasteel each is of little consequence.

The other "cost" of power armor is _time_. That's not merely strictly how long it takes to craft at a bench, but also how long it takes to collect all of the necessary materials and unlock all of the required technologies. In that sense, power armor is already fairly expensive--it requires advanced components, devilstrand, a fabrication bench, and a bunch of technologies. The time and resources to acquire all those things are considerable.

With all those things in mind, while I agree with a material cost increase, I'd scale the increases in cost back 30 - 40%.

File: Royalty/Patches/ThingDefs_Misc/Apparel_Armor.xml [hunk 1]
```diff
@@ -266,7 +280,36 @@
...
-
+	
+	<Operation Class="PatchOperationReplace">
+		<xpath>Defs/ThingDef[defName="Apparel_ArmorLocust"]/costList/Plasteel</xpath>
+		<value>
+			<Plasteel>220</Plasteel>
+		</value>
+	</Operation>
+	
+	<!-- ========== Prestige Armor ========== -->
```
Comment (N7Huntsman) [inline Royalty/Patches/ThingDefs_Misc/Apparel_Armor.xml:291]: I don't think the prestige armor is any better stat-wise than the base armor, save a psychic boost? Not enough to justify a higher construction cost, I'd say.

**Review by N7Huntsman — CHANGES_REQUESTED** [review]

File: Royalty/Patches/ThingDefs_Misc/Apparel_Armor.xml (hunk 1 above)
Comment (Tostov) [reply Royalty/Patches/ThingDefs_Misc/Apparel_Armor.xml:291]: Vanilla stuff. Dunno why, but there's an extra plasteel cost associated with prestiege armor.

File: Patches/Core/ThingDefs_Misc/Apparel_Hats.xml [hunk 2]
```diff
@@ -75,7 +75,7 @@
   <Operation Class="PatchOperationReplace">
     <xpath>Defs/ThingDef[defName="Apparel_SimpleHelmet"]/costStuffCount</xpath>
     <value>
-      <costStuffCount>50</costStuffCount>
```
[the remark this answers is not in the document]
Comment (Tostov) [reply Patches/Core/ThingDefs_Misc/Apparel_Hats.xml:78]: Typo'd. This is actually meant to be 20.

**Comment by Tostov** [comment]
> Armor cost is a bit of a conundrum; on the one hand, it's extremely important with CE--far more so than in vanilla. However, on the other hand, without a vanilla way to repair apparel, it's semi-disposable and has to be balanced as such.
> 
> Additionally, increasing the crafting costs has diminishing returns, especially for late game items such as power armor. By the point that a colony is seriously looking at mass producing power armor, an extra 50 - 60 plasteel each is of little consequence.
> 
> The other "cost" of power armor is _time_. That's not merely strictly how long it takes to craft at a bench, but also how long it takes to collect all of the necessary materials and unlock all of the required technologies. In that sense, power armor is already fairly expensive--it requires advanced components, devilstrand, a fabrication bench, and a bunch of technologies. The time and resources to acquire all those things are considerable.
> 
> With all those things in mind, while I agree with a material cost increase, I'd scale the increases in cost back 30 - 40%.

I see that, conversely, because of the mass production aspect, the cost bump is more important. If you just increase it by the 20-30% then realistically the impact is negligiable across even multiple pawns. Only by having a significant increase can you actually create some form of resource deficit from investing in producing them. When you actually have to choose between equipping two pawns with plasteel vests, or one with recon armor it becomes a lot more meaningful, because the opposite in terms of aquiring plasteel applies. I.e. by the time you have plasteel to use on armor vests, you have recon armor unlocked already.

And as noted, all power armor tier apparal practically have a slew of addtional benefits to go alongside them, most notably carry mass boosting, and the fact that most either nearly or completely offset any mass they apply to a pawn. 150-180 would be reasonable, if they didn't also provide these other benefits (for example, the ancient fallout armory heavy armor vest which does provide the same protection at about 160 cost in plasteel). But for example, an armor vest leaves a pawn with 30kg to work with, recon armor leaves a pawn with 60kg to work with, more than they originally had.

**Comment by N7Huntsman** [comment]
> Only by having a significant increase can you actually create some form of resource deficit from investing in producing them.

To a point, sure, but my thought is that drastically increasing the resource cost has a disproportionate impact on colonies that are just beginning to build their first sets of power armor by putting it further out of reach, whereas for colonies mass producing it just means they're building 3 sets instead of 4.

If we're looking to make them more difficult to acquire, I think a better option would be to find a way to increase the "time cost" that I mentioned, either by increasing the build time or requiring more of some of the more rare resources (advanced components, devilstrand). I think the time cost is something that'd be felt more uniformly across different tiers of colonies, because even being a well-established doesn't increase your _labor_ availability to nearly the same extent that it increases your _resource_ availability, if that makes any sense.=.

**Comment by N7Huntsman** [comment]
> and the fact that most either nearly or completely offset any mass they apply to a pawn.

Yes, I can appreciate that's a fairly significant benefit especially given how few ways there are to increase carryWeight and how it's linked to movement speed, but it is _power armor_--by definition, it offsets it's own weight and has a degree of self-propulsion. It makes sense to me that a pawn in 40kg of power armor will move faster than a pawn in 40kg of flak armor, for instance.

Though it may sound somewhat regressive, I might actually suggest adding back a moveSpeed penalty like it has in vanilla. This would let us slow them down a bit without decreasing their carryWeight. I'd do some experimenting and set the penalty such that, all other things being equal, a pawn in "marine" power armor loaded for bear is about the same speed or _slightly_ slower than a rifleman with a reasonable loadout. Recon armor and other light variants should be somewhat faster, heavier variants should be somewhat slower. Thoughts?

File: Royalty/Patches/ThingDefs_Misc/Apparel_Armor.xml (hunk 1 above)
Comment (N7Huntsman) [reply Royalty/Patches/ThingDefs_Misc/Apparel_Armor.xml:291]: Weird. Alright, well in vanilla the cost is 20% more plasteel, but I wouldn't make it that much for CE. I'd just do like 5 - 10%, depending on the final cost--a fairly token amount.

**Comment by Tostov** [comment]
> To a point, sure, but my thought is that drastically increasing the resource cost has a disproportionate impact on colonies that are just beginning to build their first sets of power armor by putting it further out of reach, whereas for colonies mass producing it just means they're building 3 sets instead of 4 (which sure, is a 25% decrease, but the gulf in capability between having 0 sets of power armor and 1 set is a hell of lot wider than having 3 instead of 4).
> 
> If we're looking to make them more difficult to acquire, I think a better option would be to find a way to increase the "time cost" that I mentioned, either by increasing the build time or requiring more of some of the more rare resources (advanced components, devilstrand). I think the time cost is something that'd be felt more uniformly across different tiers of colonies, because even being a well-established doesn't increase your _labor_ availability to nearly the same extent that it increases your _resource_ availability, if that makes any sense.=.

The thing with advanced components, compared to plasteel, is that they're a lot more granular in terms of how much each one adds, and in a way they draw from a seperate general resource pool to plasteel. And to me, that core issue really is the ability to get those first sets of power armor so easily, which is what makes them so desirable to just rush instead of doing anything else with in the first place. The plasteel efficency is key in determining whether it's worth using it for power armor, since the moment that it becomes too efficent, it becomes massively more desirable to skip using it on anything other than power armor, since that's what you want to save it for. Adding more advanced component cost doesn't make power armor nearly as less appealing compared to expending that plasteel on other items preciesly because they're different general resource pools.

You can only buy so much plasteel from each trader. If you split the difference across advanced components and plasteel, the efficency of plasteel use doesn't change as much since you have the offset from getting easier advanced components and not having to fabricate all of them yourself. To me, a well estabilished colony that can pump out power armor is most likely going to yes, have enough plasteel to pump out power armor, but that sort of resource/production sink is going to generally raise the value of the colony by quite a bit, in addtion to the extra armor that each pawn is wearing.

But yeah, to me the exact scenario I want to induce is that strong question of whether you want to really scrape together and get your first sets of power armor, or invest across multiple pawns with conventional armor vests instead. In essence, that jump from plasteel armor vests to actual power armor should feel like a jump and not just an easy step.

**Review by N7Huntsman — APPROVED** [review]

**Pull request merged**